sha512_mb/sha512_mb_test.c

```diff
@@ -118,6 +118,78 @@ static uint64_t *expResultDigest[MSGS] = { expResultDigest1, expResultDigest2,
 	expResultDigest7, expResultDigest8
 };
 
+#define NUM_CHUNKS	4
+#define DATA_BUF_LEN	512
+#define CSUM_BUF_LEN	64
+int non_blocksize_updates_test(SHA512_HASH_CTX_MGR * mgr)
+{
+	SHA512_HASH_CTX ctx_refer;
+	SHA512_HASH_CTX ctx_pool[NUM_CHUNKS];
+	SHA512_HASH_CTX *ctx = NULL;
+
+	const int update_chunks[NUM_CHUNKS] = { 32, 64, 128, 256 };
+	unsigned char data_buf[DATA_BUF_LEN];
+	unsigned char refer_csum_buf[CSUM_BUF_LEN];
+
+	memset(data_buf, 0xA, DATA_BUF_LEN);
+
+	// Init contexts before first use
+	hash_ctx_init(&ctx_refer);
+
+	ctx = sha512_ctx_mgr_submit(mgr, &ctx_refer, data_buf, DATA_BUF_LEN, HASH_ENTIRE);
+	if (ctx && ctx->error) {
+		return -1;
+	}
+	ctx = sha512_ctx_mgr_flush(mgr);
+	if (ctx && ctx->error || ctx_refer.status != HASH_CTX_STS_COMPLETE) {
```

Comment:
getting a warning here on gcc and clang so should add an extra (), perhaps on both for clarity.

    if ((ctx && ctx->error) || (ctx_refer.status != HASH_CTX_STS_COMPLETE)) {